### PR TITLE
[glsl-in] improved WIP parsing for declarations

### PR DESCRIPTION
**Pull request by Imberflur**

### CODE REVIEW

File: src/front/glsl/ast.rs [hunk 1]
```diff
@@ -301,10 +301,12 @@ impl ExpressionRule {
 #[derive(Debug)]
 pub enum TypeQualifier {
     StorageQualifier(StorageQualifier),
+    Interpolation(Interpolation),
     ResourceBinding(ResourceBinding),
     Binding(Binding),
-    Interpolation(Interpolation),
     Sampling(Sampling),
+    Std140,
```
Comment (JCapucho) [inline src/front/glsl/ast.rs]: We should name this to something like `Layout(StructLayout)` and have
```
enum StructLayout {
  Std140
  // These can be added later
  Std430
  Shared
  Packed
}
```

File: src/front/glsl/error.rs [hunk 2]
```diff
@@ -45,6 +46,9 @@ impl fmt::Display for ErrorKind {
             ErrorKind::UnknownField(ref meta, ref val) => {
                 write!(f, "Unknown field {} at {:?}", val, meta)
             }
+            ErrorKind::UnknownLayoutQualifier(ref meta, ref val) => {
+                write!(f, "Unknown layout qualifier name {} ad {:?}", val, meta)
```
Comment (JCapucho) [inline src/front/glsl/error.rs]: typo `ad` -> `at`

File: src/front/glsl/parser.rs [hunk 3]
```diff
@@ -46,6 +50,15 @@ impl<'source, 'program, 'options> Parser<'source, 'program, 'options> {
         self.lexer.next().ok_or(ErrorKind::EndOfFile)
     }
 
+    fn bump_if(&mut self, value: TokenValue) -> Option<Token> {
+        // Ignore end of file
+        if self.lexer.peek().map_or(false, |t| t.value == value) {
+            self.bump().ok()
```
Comment (JCapucho) [inline src/front/glsl/parser.rs:56]: We can do something like
```rust
self.lexer.peek().filter(|t| t.value == value).and_then(|_| self.bump().ok())
```

**Review by JCapucho — COMMENTED** [review]
great job, i have left some comments but once you fix the clippy issues we can merge

File: src/front/glsl/ast.rs (hunk 1 above)
Comment (Imberflur) [reply src/front/glsl/ast.rs]: we already have a type with this name :grimacing: 
```rust
#[derive(Debug, Clone)]
pub enum StructLayout {
    Binding(Binding),
    Resource(ResourceBinding),
    PushConstant,
}
```

File: src/front/glsl/parser.rs (hunk 3 above)
Comment (Imberflur) [reply src/front/glsl/parser.rs:56]: the borrow checker can't see through this :slightly_frowning_face: 
```
   |
55 |         self.lexer
   |         ---------- borrow occurs here
...
58 |             .and_then(|_| self.bump().ok())
   |              -------- ^^^ ---- second borrow occurs due to use of `self` in closure
   |              |        |
   |              |        closure construction occurs here
   |              first borrow later used by call
```

Comment (Imberflur) [reply src/front/glsl/parser.rs:56]: oh sweet! https://doc.rust-lang.org/std/iter/struct.Peekable.html#method.next_if

**Pull request merged**